services/alertmanager/registry.go

```diff
@@ -55,6 +56,7 @@ func (r *Registry) CreateAlert(id string, labels, annotations map[string]string,
 			Labels: labels,
 		},
 
+		EndsAt: strfmt.DateTime(now().Add(resolveTimeoutFactor * resendInterval)),
 		// StartsAt and EndAt can't be added there without changes in Registry
```

Comment:
Actually, that comment is correct – `delayFor` will not work correctly with `EndsAt` without modifications.